br/pkg/lightning/config/config.go

```diff
@@ -1130,6 +1188,21 @@ func (cfg *Config) AdjustCommon() (bool, error) {
 }
 
 func (cfg *Config) CheckAndAdjustForLocalBackend() error {
+	if cfg.TikvImporter.IncrementalImport {
+		switch cfg.TikvImporter.AddIndexBySQL {
+		case BoolTrue:
+			return common.ErrInvalidConfig.
+				GenWithStack("tikv-importer.add-index-using-ddl cannot be used with tikv-importer.incremental-import")
+		case BoolAuto:
+			cfg.TikvImporter.AddIndexBySQL = BoolFalse
+		}
+	} else {
+		// Automatically enable add-index-by-sql if failpoint is enabled. (For integration test)
+		if cfg.TikvImporter.AddIndexBySQL == BoolAuto && common.IsFailpointBuild {
```

Comment:
prefer add an explicit failpoint to enable it when needed

Reply:
This is a temporary plan for all integration tests. Because TiDB v7.0.0 has not been released yet and we want to enable this feature for TiDB v7.0.0 by default.

After TiDB v7.0.0 release, we can remove this code snippet.